src/boot/efi/pe.c

```diff
@@ -151,9 +151,17 @@ static void locate_sections(
         assert(offsets);
         assert(sizes);
 
+        size_t prev_section_addr = 0;
+
         for (UINTN i = 0; i < n_table; i++) {
                 const PeSectionHeader *sect = section_table + i;
 
+                if (in_memory) {
+                        if (prev_section_addr > sect->VirtualAddress)
+                                log_error_stall(u"Overlapping PE sections detected. Boot may fail due to image memory corruption!");
+                        prev_section_addr = sect->VirtualAddress + sect->VirtualSize;
```

Comment:
maybe if we are at it also do an overflow check?

Reply:
The image is already in memory, relocated and running. There's hardly any point in doing that now.

Reply:
you never know what quirks the firmware has. As a matter of hygiene I'd always check things like that in new code.

Reply:
This has never been an issue so I really fail to see the point in doing so. This was just supposed to be a safety measure to detect badly assembled unified images that are already running and you wanna make me jump through pointles hoops? nope. Use this or stick your head in the sand.

And no, detecting this so the image never even gets shown in the boot menu is a terrible idea. People will complain that systemd-boot is too stupid to pick up its own menu entries.

Reply:
dunno, it's just a matter of hygiene. For new code we try to be more careful than we used to and always assume that something like a supercharged `-ftrapw` mode (that also covers unsigned ints) was in effect.

Reply:
(i.e. clang has "intsan" for that as i understand)